Gordon360/Services/HousingService.cs

```diff
@@ -0,0 +1,774 @@
+
+
+            //--------
+            // Update applicant information
+
+            IEnumerable<GET_AA_APPLICANTS_BY_APPID_Result> existingApplicantResult = null;
+
+            appIDParam = new SqlParameter("@APPLICATION_ID", applicationID);
+
+            // Get the IDs of the applicants that are already stored in the database for this application
+            existingApplicantResult = RawSqlQuery<GET_AA_APPLICANTS_BY_APPID_Result>.query("GET_AA_APPLICANTS_BY_APPID @APPLICATION_ID", appIDParam);
+            if (existingApplicantResult == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "The applicants could not be found." };
+            }
+
+            // List of applicants IDs that are in the array recieved from the frontend but not yet in the database
+            List<ApartmentApplicantViewModel> applicantsToAdd = new List<ApartmentApplicantViewModel>(newApartmentApplicants);
+
+            // List of applicants IDs that are in both the array recieved from the frontend and the database
+            List<ApartmentApplicantViewModel> applicantsToUpdate = new List<ApartmentApplicantViewModel>();
+
+            // List of applicants IDs that are in the database but not in the array recieved from the frontend
+            List<ApartmentApplicantViewModel> applicantsToRemove = new List<ApartmentApplicantViewModel>();
+
+            // Check whether any applicants were found matching the given application ID number
+            if (existingApplicantResult.Any())
+            {
+                foreach (GET_AA_APPLICANTS_BY_APPID_Result existingApplicant in existingApplicantResult)
+                {
+                    ApartmentApplicantViewModel newMatchingApplicant = null;
+                    newMatchingApplicant = newApartmentApplicants.FirstOrDefault(x => x.StudentID == existingApplicant.ID_NUM);
+                    if (newMatchingApplicant != null)
+                    {
+                        // If the applicant is in both the new applicant list and the existing applicant list, then we do NOT need to add it to the database
+                        applicantsToAdd.Remove(newMatchingApplicant);
+                        if (newMatchingApplicant.OffCampusProgram != existingApplicant.AprtProgram)
+                        {
+                            // If the applicant is in both the new and existing applicant lists but has different OffCampusProgram values, then we need to update that in the database
+                            applicantsToUpdate.Add(newMatchingApplicant);
+                        }
+                    }
+                    else
+                    {
+                        // If the applicant is in the existing list but not in the new list of applicants, then we need to remove it from the database
+                        applicantsToRemove.Add(newMatchingApplicant);
```

Comment:
This still needs to be fixed. `newMatchingApplicant` is necessarily null in this branch of the `if`. You mean to add the `existingApplicant`, but the type mismatch complicates that slightly. 

This error makes it impossible to remove anyone from an already saved application currently.